connect/src/spirc.rs

```diff
@@ -595,24 +613,32 @@ impl SpircTask {
                 }
             }
             SpircCommand::VolumeUp => {
+                self.handle_volume_up();
```

Comment:
Should we handle volume events if we are not the active device?

Reply:
Prob not.

Reply:
Now that I really think about it we really shouldn't handle any events at all if we're not active. We should just forward them.

Reply:
Actually should we even forward them?

Reply:
No, forwarding is not necessary. `spirc` is like a chat room, all clients and devices get all messages. Hence "IRC" I think.

Reply:
I'm thinking we should ignore all commands except shutdown if we're not active.

Reply:
> No, forwarding is not necessary. spirc is like a chat room, all clients and devices get all messages. Hence "IRC" I think.

The commands don't some from other clients they come from us. I'm not sure why they forward commands now while doing nothing to spirc's internal state if we're not active. They should be ignored if we're not active (as not to fire a command unintentionally when we're not active).

Reply:
There is a catch-all in the Mercury handler which checks whether the messages were directed at us. So this now never happens. Wondering though what this `active` check does here — is it redundant or am I forgetting something? On mobile now but should do a code walkthrough to make sure.

Reply:
From what I gather sending a command explicitly pulls in our state (the `self` part) so forwarding commands doesn't make much sense. If the point of the public functions in spirc is to allow people to use spirc to control anything but librespot they are broken because of that. If the intention is to allow other things to hook into spirc and control just librespot then forwarding commands when we're not active is at best redundant and at worse can lead to an inconsistent state.

Reply:
That catch-all essentially makes spirc a one trick pony. It restricts what spirc can control. It makes it so spirc can only control 1 device ident.

Reply:
I agree with you for the `Connect` crate it’s obvious. If anyone wants to tap into it anyway they still do it quite simply be registering a listener in `core/session`. 

But it was not the point of my question/comment. What I want to make sure is why we have an `active` book when the listener already checks whether messages are directed at us. There might be a good reason, there might not.

Reply:
I don't really think there is. If a single instance of spirc is only ever meant to deal with one device ident (I was not sure if that was the intention? Because that's how it works now) then I'm going to go back to my statement that we should ignore all commands except shutdown in `handle_command`.

Reply:
*ignore them if we're not active

Reply:
Sending commands to Spotify other than goodbye (shutdown) when we're not active is just noise and wasted API calls.

Reply:
I agree but did you do the code walkthrough and test just to see if we are overlooking something?

Reply:
I did but another set of eyes who understands what's going on better than me sure couldn't hurt. There's no huge hurry with this. I'm Ok with taking our time.

Reply:
I should say I did the code walk though. As far as testing goes, there's nothing to test nothing in librespot uses the public functions that call the commands currently and it not easy to get a hold of spirc since it's not currently cloneable so they're basically dead code. A better approach would be to expose a command channel much like player exposes the event channel (but in reverse ofc).

Reply:
The alternative is to make it cloneable but then you'd need a fresh clone every time you wanted to call one of the functions.

Reply:
There I fixed the commands. Now all commands are ignored except Shutdown if we're not active as it should be.